telegram/ext/_conversationhandler.py

```diff
@@ -781,7 +785,7 @@ async def handle_update(  # type: ignore[override]
             else:
                 block = DefaultValue.get_value(handler.block)
 
-        try:
+        try:  # Now determine if the handler in the current state should be handled asynchronously
```

Comment:
no, we already determined that above. we now just act depending on the result ;)